### PR TITLE
Add base documentation

### PR DESCRIPTION
**Pull request by danlamanna**
Addresses #35 and #77

Nothing to review here yet @kotfic, but we can use this PR as a means of discussing a TOC and collaborating on docs.

I'm thinking a baseline TOC could look something like this:
```
* What is Geonotebook
** Goals
** Architecture
* Installation
** Basic Install
** Development Install
** Vagrant
** Docker
* User Documentation
** Walkthrough (example notebooks)
* Developer Documentation
** API Docs
** Adding an RPC call
*** Server Side
*** Client Side
```

### CODE REVIEW

**Comment by danlamanna** [comment]
Part of this PR should be adding sphinx docs to each of the user facing commands on the `M` object so `M.set_center?` (for example) within the notebook will provide good docs.

**Comment by danlamanna** [comment]
@jbeezley PTAL

Note once this is merged, @dorukozturk can delete the S3 bucket we were previously relying on for geotiffs.

File: docs/source/developer-docs.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+.. figure:: https://docs.google.com/drawings/d/1Zss2O5rXANNYnaooZhnKhWCVu6wjURmBWNeKLL_QXXc/pub?w=960&h=720
+   :align: center
+
+   Components of the Jupyter Notebook
+
+When you execute python in a Jupyter Notebook cell the python code was transmitted (as a string) over a web socket to the webserver, then proxied through ZeroMQ to the IPykernel where it was evaluated as a Python expression. The results of that execution are then sent back over the ZeroMQ/Websocket connection and displayed in your browser. This is the standard way in which Jupyter notebook takes code from a web browser, and executes it in an interactive shell (kernel) and returns the output.
+
+
+*TODO introduce comm channels*
```
Comment (aashish24) [inline docs/source/developer-docs.rst]: @danlamanna are we planning to keep TODO's?

Comment (danlamanna) [inline docs/source/developer-docs.rst]: We can, or we can put them inside an issue with a checklist. It's up to us.

Comment (aashish24) [inline docs/source/developer-docs.rst]: My slight preference to put them inside an issue with a checklist if it is not too much work but I am okay with this as it is (if you want to go ahead and merge it).

**Comment by danlamanna** [comment]
related: #111

**Pull request merged**